File: src/commands/graph.ts
import { Command } from "commander";
import { request } from "../lib/client.js";
import { outputSuccess } from "../lib/output.js";

export function registerGraphCommand(program: Command): void {
  program
    .command("graph <libraryId>")
    .description("Get the full document graph for a library")
    .action(async (libraryId: string) => {
      const data = await request(`/libraries/${libraryId}/graph`);
      outputSuccess(data);
    });
}

File: src/commands/graph.rs
//! `graph <libraryId>`: the full document graph for a library.

use crate::client::request;
use crate::error::CliError;
use crate::output;

pub fn run(library_id: &str) -> Result<(), CliError> {
    let data = request(&["libraries", library_id, "graph"], &[])?;
    output::success(&data);
    Ok(())
}
